### PR TITLE
Fetch report images for WeasyPrint via Plone Subrequests

### PR DESCRIPTION
**Pull request by ramonski**
## Description of the issue/feature this PR addresses

This Pull Requests supersedes https://github.com/senaite/senaite.core/pull/539

Linked issue: https://github.com/senaite/senaite.core/issues/538

The current provided PR uses now [Plone Subrequest](https://github.com/plone/plone.subrequest) to fetch the data directly. Also see: https://pypi.python.org/pypi/plone.subrequest

This bypasses the need to hack the authentication into the request or fetch the file data by hand, e.g. from an Attachment.

## Current behavior before PR

Images do not appear in the final PDF report

## Desired behavior after PR is merged

Images are displayed on PDF reports

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: bika/lims/utils/__init__.py [hunk 1]
```diff
@@ -382,31 +389,41 @@ def isnumber(s):
         return False
 
 
-def bika_url_fetcher(url):
+def senaite_url_fetcher(url):
```
Comment (xispa) [inline bika/lims/utils/__init__.py]: Just update the comment, we are not using `__ac` cookie any more

File: bika/lims/utils/__init__.py [hunk 2]
```diff
@@ -382,31 +389,41 @@ def isnumber(s):
...
+    context = portal.restrictedTraverse(path, None)
+
+    if context is None:
+        logger.info("URL is external, passing over to the default URL fetcher...")
+        return default_url_fetcher(url)
+
+    logger.info("URL is local, fetching data by path '{}' via subrequest".format(path))
+
+    # get the data via subrequest
+    response = subrequest(path)
```
Comment (ramonski) [inline bika/lims/utils/__init__.py]: Here I'm not sure if I should hand in the URL directly or the physical path. I didn't yet analyse the code from p.s. to see if this makes a change at all. If not, we could probably remove all the URL-Path conversion from above...

Comment (xispa) [reply bika/lims/utils/__init__.py]: As far as I see in the examples from https://pypi.python.org/pypi/plone.subrequest , they are always using the path, never the full URL. But well, we can inspect that `subrequest` function

Comment (xispa) [reply bika/lims/utils/__init__.py]: Maybe you are right, see this:
https://github.com/plone/plone.subrequest/blob/8b5cd141b3a2adc3d1dad381376c7e5adc80da22/plone/subrequest/__init__.py#L103-L114

**Pull request merged**